src/ubx-message.toit

```diff
@@ -200,90 +503,833 @@ class AckAck extends Message:
+// Note:  M8 and M9 (and later versions of 6M have extra data returned
+//        in extensions.
+class MonVer extends Message:
+  /** The UBX-MON-VER message ID. */
+  static ID ::= 0x04
+
+  /** Construct a poll-request UBX-MON-VER. */
+  constructor.poll:
+    super.private_ Message.MON ID #[]
+
+  /** Construct from an incoming payload. */
+  constructor.private_ payload/ByteArray:
+    super.private_ Message.MON ID payload
+
+  /** See $super. */
+  id-string_ -> string:
+    return "VER"
+
+  /** Software version string (NULL-terminated inside 30 bytes). */
+  sw-version -> string:
+    return convert-string_ 0 30
+
+  /** Hardware version string (NULL-terminated inside 10 bytes). */
+  hw-version -> string:
+    return convert-string_ 30 10
+
+  /** Returns true if an extension row exists containing the supplied string. */
+  has-extension str/string -> bool:
+    //return (extensions-raw.contains extension-name)
+    return (extensions-raw.any: (it.index-of str) > -1)
+
+  /** If a string exists in the extensions with the supplied text, return the
+  entire line . */
+  extension str/string -> string?:
+    //return (extensions-raw.contains extension-name)
+    extensions-raw.any:
+      if (it.index-of str) > -1:
+        return it
+    return null
+
+  /**
+  Returns a map of extension string AVPs.
+
+  This function returns a map of strings with the keyed by the first part, with
+    the value being the remainder past the first instance of "=".
+
+  DISABLED: Originally it looked like the extensions would be AVPs delimited by
+    '='.  This is not true in all device generations. In addition, the sets of
+    tags shown in the extensions aren't grouped, and can be split across more
+    than one extension 'row'.  So for now, this is disabled in favor of the
+    driver doing what is required for parsing of these messages.
+
+  extensions -> Map:
+    raw-extensions := extensions-raw
+    output-extensions := {:}
+    print "TESTING $(extensions-raw)"
+    raw-extensions.do:
+      eq-pos := it.index-of "="
+      if eq-pos > -1:
+        output-extensions[it[..eq-pos]] = it[..(eq-pos + 1)]
+      else:
+        output-extensions[it] = ""
+    return output-extensions
+  */
+
+  /**
+  Returns a list of extension strings, if present.
+
+  If provided by the firmware version on the device, it provides a list of n 30
+    byte entries.  Each entry is a NUL-terminated ASCII string with an AVP
+    delimited by '='.
+  */
+  extensions-raw -> List:
+    raw-extensions := []
+    offset := 40
+    eq-pos := ?
+    while offset + 30 <= payload.size:
+      raw-extensions.add (convert-string_ offset 30)
+      offset += 30
+    return raw-extensions
+
+
+  /** Helper: read a NULL-terminated string from a fixed-size field. */
+  convert-string_ start length -> string:
+    // Find first NUL within [start .. start+length).
+    end := start
+    limit := start + length
+    while (end < limit) and (LITTLE-ENDIAN.uint8 payload end) != 0:
```

Comment:
Here I would not use `LITTLE-ENDIAN.uint8` but index directly.

In theory you don't need to check against the limit. The string is supposed to be null-terminated.